spec/api/v1/answers_spec.rb

```diff
@@ -7,70 +7,44 @@
-        expect(response.status).to eq 401
-      end
+    it_behaves_like 'API index when authorized', %w[id body created_at updated_at] do
+      let(:object) { answer }
+      let(:size) { 2 }
+      let(:path) { '0/' }
     end
 
-    context 'when authorized' do
-      before do
-        create(:answer, question: question, user: user)
-        get "/api/v1/questions/#{question.id}/answers", params: { format: :json, access_token: access_token.token }
-      end
-
-      it { expect(response).to be_success }
-      it { expect(response.body).to have_json_size(2) }
-
-      %w[id body created_at updated_at].each do |attr|
-        it { expect(response.body).to be_json_eql(answer.send(attr.to_sym).to_json).at_path("0/#{attr}") }
-      end
+    def do_request(options = {})
+      get "/api/v1/questions/#{question.id}/answers", params: { format: :json }.merge(options)
     end
 
-    context 'when authorized invalid path' do
-      subject(:parsed_response) { JSON.parse(response.body) }
-
-      before do
-        create(:answer, question: question, user: user)
-        get "/api/v1/questions/#{question.id + 2}/answers", params: { format: :json, access_token: access_token.token }
-      end
+    def invalid_path
+      get "/api/v1/questions/#{question.id + 2}/answers", params: { format: :json, access_token: access_token.token }
+    end
 
-      it { expect(parsed_response['status']).to eq('error') }
-      it { expect(parsed_response['code']).to eq(404) }
-      it { expect(parsed_response['message']).to eq("Can't find question") }
+    def index_authorized
+      create(:answer, question: question, user: user)
+      get "/api/v1/questions/#{question.id}/answers", params: { format: :json, access_token: access_token.token }
     end
   end
 
   describe 'GET /show' do
-    context 'when unauthorized' do
-      it 'return 401 status if there is no access_token' do
-        get "/api/v1/answers/#{answer.id}", params: { format: :json }
-        expect(response.status).to eq 401
-      end
+    it_behaves_like 'API Authenticable'
+    it_behaves_like 'API show invalid path'
 
-      it 'return 401 status if access_token is valid' do
-        get "/api/v1/answers/#{answer.id}", params: { format: :json, access_token: '1234' }
-        expect(response.status).to eq 401
-      end
+    it_behaves_like 'API index when authorized', %w[id body created_at updated_at] do
+      let(:object) { answer }
+      let(:size) { 7 }
+      let(:path) { '' }
     end
 
     context 'when authorized' do
       let!(:comment) { create(:comment, commentable: answer, user: user) }
 
       before { get "/api/v1/answers/#{answer.id}", params: { format: :json, access_token: access_token.token } }
 
-      it { expect(response).to be_success }
-
-      %w[id body created_at updated_at].each do |attr|
-        it { expect(response.body).to be_json_eql(answer.send(attr.to_sym).to_json).at_path(attr.to_s) }
-      end
-
       context 'with comment' do
```

Comment:
Is it possible https://github.com/ValAntG/Hard_course/blob/73e68266b476f87bf075a898e2a9b5167e9a6a3f/spec/api/v1/answers_spec.rb#L48-L53
take out here
https://github.com/ValAntG/Hard_course/pull/16/files#diff-72b981c01cb01ac2f90ce7664f34fbd0
?